terraform/gt-site.tf

```diff
@@ -0,0 +1,91 @@
+provider "aws" {
+  profile = "${var.profile}" # Assumes you have valid AWS credentials.
+  region  = "${var.region}"
+}
+
+# --- ECS RESOURCES --- #
+
+resource "aws_ecs_cluster" "gt-site-cluster" {
+  name = "gt-site-cluster"
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+# `containers.json` is analogous to `docker-compose.yml`.
+# All the container settings are there.
+resource "aws_ecs_task_definition" "gt-site-cluster-task" {
+  family                = "gt-site-cluster"
+  container_definitions = "${file("containers.json")}"
+}
+
+resource "aws_ecs_service" "gt-site-cluster-service" {
+  name                               = "gt-site-cluster"
+  cluster                            = "${aws_ecs_cluster.gt-site-cluster.id}"
+  task_definition                    = "${aws_ecs_task_definition.gt-site-cluster-task.family}:${aws_ecs_task_definition.gt-site-cluster-task.revision}"
+  desired_count                      = "${var.desired_instance_count}"
+  iam_role                           = "${var.ecs_service_role}"
+  deployment_minimum_healthy_percent = "0"
+  deployment_maximum_percent         = "100"
+
+  load_balancer {
+    elb_name       = "${aws_elb.gt-site-elb.name}"
+    container_name = "gtsite-nginx"
+    container_port = 8080
+  }
+}
+
+# A load balancer is necessary to launch the EC2 manager instance properly.
+resource "aws_elb" "gt-site-elb" {
+  subnets = ["${var.subnet_id}"]
+
+  listener {
+    lb_port           = 80
+    lb_protocol       = "HTTP"
+    instance_port     = 8080   # The port that the nginx container is listening on.
+    instance_protocol = "HTTP"
+  }
+
+  cross_zone_load_balancing = false
+  idle_timeout              = 3600
+
+  tags {
+    Name = "gt-site-cluster elb"
+  }
+}
+
+resource "aws_launch_configuration" "ecs" {
```

Comment:
Reusing the default VPC, the value for `vpc_id` would be `vpc-617f9604`.

Despite being marked as optional, that isn't realistically true. Prior to the existence of VPCs there was what is now known as "EC2-Classic". This environment gave each EC2 instance a public IP (and you had no other choice).

Not supplying `vpc_id` for a security group makes it so that the security group is an EC2-Classic security group, which isn't compatible with EC2-VPC security groups.

---

The other security group ID is referencing the security group for the load balancer itself. That's something like: https://github.com/azavea/raster-foundry-deployment/blob/develop/terraform/app.tf#L4-L12